File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_sum = curr_sum = sum(nums[:k])
        for r in range(k, len(nums)):
            curr_sum += nums[r]
            curr_sum -= nums[r - k]
            max_sum = max(max_sum, curr_sum)
        return max_sum / k

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double win_sum = std::accumulate(nums.begin(), nums.begin() + k, 0);
        double max_sum = win_sum;
        for (int r = k; r < nums.size(); ++r) {
            win_sum += nums[r];
            win_sum -= nums[r - k];
            max_sum = std::max(max_sum, win_sum);
        }
        return max_sum / k;
    }
};